paasta_tools/cli/cmds/validate.py

```diff
@@ -187,17 +198,29 @@ def no_duplicate_instance_names_message(service, cluster):
     return success(f"All {service}'s instance names in cluster {cluster} are unique")
 
 
-def get_schema(file_type):
+def get_schema_validator(file_type: str) -> Draft4Validator:
     """Get the correct schema to use for validation
 
     :param file_type: what schema type should we validate against
     """
-    schema_path = "schemas/%s_schema.json" % file_type
-    try:
-        schema = pkgutil.get_data("paasta_tools.cli", schema_path).decode()
-    except IOError:
-        return None
-    return json.loads(schema)
+    schema_path = f"schemas/{file_type}_schema.json"
+    autoscaling_path = "schemas/autoscaling_schema.json"
+    schema = pkgutil.get_data("paasta_tools.cli", schema_path).decode()
+    autoscaling_ref = pkgutil.get_data("paasta_tools.cli", autoscaling_path).decode()
+
+    base_schema = json.loads(schema)
+    autoscaling_schema = json.loads(autoscaling_ref)
+    store = {
+        "base": base_schema,
+        autoscaling_schema["$id"]: json.loads(autoscaling_ref),
+    }
```

Comment:
what does this end up doing for schemas like tron's - where autoscaling doesn't exist?

Reply:
This is just setting up a cacheable "store" of schemas so that you can reference them later on.  If you don't actually reference anything (like in the tron case) then it does nothing.

I added a comment to this effect.